app/models/concerns/audit_associations.rb

```diff
@@ -19,56 +15,53 @@ def find_association_class(name)
     end
 
     def associated_changes
-      associations = Array.wrap(audited_options[:associations])
-      associations.each_with_object({}) do |association_name, changes_hash|
-        association_ids = "#{association_name.to_s.singularize}_ids"
-        if send("#{association_ids}_changed?")
-          association_class = find_association_class(association_name)
-          change = send("#{association_ids}_change")
-          change_ids = change.flatten.uniq
-
-          id_name_map = association_class.where(id: change_ids).inject({}) { |r, p| r.merge(p.id => p.to_label) }
+      audited_options[:associations].each_with_object({}) do |association, changes|
+        association_ids = "#{association.to_s.singularize}_ids"
+        if public_send("#{association_ids}_changed?")
+          change = public_send("#{association_ids}_change")
 
-          changes_hash[association_name] = change.each_with_object([]) do |ids, humaized_associations|
-            humaized_associations << ids.each_with_object([]) do |id, humanized_ids|
-              humanized_ids << id_name_map[id]
-            end.join(', ')
+          changes[association] = change.map do |ids|
+            associated_names(association, ids)
           end
         end
       end
     end
-  end
 
-  module Auditor
-    def self.included(base)
-      base.extend ClassMethods
+    def associated_attributes
+      audited_options[:associations].each_with_object({}) do |association, attributes|
+        ids = public_send("#{association.to_s.singularize}_ids")
+        attributes[association] = associated_names(association, ids)
+      end
     end
 
-    module ClassMethods
-      include Audited::Auditor::ClassMethods
-
-      def audited(options = {})
-        if options[:associations].present?
-          configure_dirty_associations(Array(options[:associations]))
-        end
+    def associated_names(association, ids)
+      find_association_class(association).where(id: ids).map(&:to_label).to_sentence
+    end
+  end
 
-        super
+  module AssociationsDefinitions
+    def audited(options = {})
+      options[:associations] = Array(options[:associations])
```

Comment:
Normalizing the option to an array here (even if it's nil - so it's turned to empty array) simplifies handling it in other places.

Reply:
There is a PR open in audited to change associated_with to be `has_many` instead of `has_one`. There still isn't support there for tracking changes in associations. (but PRs welcome :wink: )

Reply:
I meant for existing options in audited, but I see that's already happening https://github.com/collectiveidea/audited/blob/master/lib/audited/auditor.rb#L363-L368